osf/utils/migrations.py

```diff
@@ -419,6 +419,7 @@ def state_forwards(self, app_label, state):
 
     def database_forwards(self, app_label, schema_editor, from_state, to_state):
         ensure_schemas(to_state.apps)
+        map_schemas_to_schemablocks(to_state.apps)
```

Comment:
Another concern is that [`map_schemas_to_schema_blocks` will always delete existing schema blocks](https://github.com/CenterForOpenScience/osf.io/blob/1dde647bfca24b6034d35d0d5bc9baa78cf4e32c/osf/utils/migrations.py#L388-L389) -- we probably want to leave any existing blocks alone (they may have been updated).  

It may be best to update `map_schemas_to_schema_blocks` to skip schemas that already have blocks. There's only [one usage](https://github.com/CenterForOpenScience/osf.io/blob/04dba4419033520c9b1f80fa5b7a2485c237997e/osf/migrations/0191_migrate_schemas_to_schemablocks.py#L73) and it seems safe enough to have that updated behavior there, too (since the schemablock table is added fresh in the prior migration, and it'll delete schema blocks in the reverse migration).

Reply:
So, on staging2 we'll need to back up to before `0191_migrate_schemas_to_schemablocks` and re-migrate forward to get the schema updates to be reflected in schema blocks?

Reply:
or just manually drop all the existing blocks?

Reply:
i guess so -- that or manually delete the schema blocks.

i could also be wrong about what we want -- it might be better to assume we'll exclusively update schema blocks via this migration operation, so we *do* want to delete all existing blocks each time. That would fit with eventually replacing those `website/project/metadata/*.json` files with schema block representations, which would be loaded as-is, replacing any existing schema blocks.

Reply:
which is probably what we'll need to do because we'll need to run `ensure_schemas` *before* any schema blocks are created and we probably don't want to back all the way up to `0173_ensure_schemas`. Or we could edit `0191_migrate_schemas_to_schemablocks` and have it `ensure_schemas` first?

Reply:
Yeah, I'm not sure of what we'll want in the future. For now they have to be synced, right?

Reply:
Yeah, I think we'll want to have one way of updating a schema at any given time. So if currently it's to update the json, then let's keep deleting the schemablocks when mapping the schemas. If we ever go to updating schemablocks directly, then we can eliminate updating the schema json as a valid path.